lib/rust/frp2/src/nodes.rs

```diff
@@ -0,0 +1,250 @@
+//! FRP nodes definitions. They are the main user-facing API for creating FRP networks.
+
+use crate::prelude::*;
+
+use crate::data::Data;
+use crate::network::Network;
+use crate::node::input::Listen;
+use crate::node::input::ListenAndSample;
+use crate::node::input::Sample;
+use crate::node::Node;
+use crate::node::NodeInNetwork;
+use crate::node::NodeWithDefaultOutput;
+use crate::node::TypedNode;
+
+
+
+// ==============
+// === Stream ===
+// ==============
+
+/// Marker type for the [`Stream`] node.
+#[derive(Clone, Copy, Debug, Default)]
+pub struct STREAM;
+
+auto trait NotStream {}
+impl !NotStream for STREAM {}
+
+/// The most generic type of a node. Every node can be converted to this type while losing some
+/// abilities. For example, a [`Sampler`] node provides the ability to sample the last node output.
+/// However, if you need to store different node types in an array, you can convert them to samplers
+/// with zero-overhead.
+pub type Stream<Output = ()> = TypedNode<STREAM, Output>;
+
+impl<Type: NotStream, Output> Deref for TypedNode<Type, Output> {
+    type Target = Stream<Output>;
+    #[inline(always)]
+    fn deref(&self) -> &Self::Target {
+        // # Safety
+        // The values differ only in phantom parametrization, so it is safe to coerce them.
+        #[allow(unsafe_code)]
+        #[allow(trivial_casts)]
+        unsafe {
+            &*(self as *const Self as *const Self::Target)
+        }
+    }
+}
```

Comment:
I added `as_stream` method. However, I left `Deref` there because it allows for a very nice pattern. If I have a node of any type `n` and a function `f` which accepts stream, I can do `f(&n)`, as DerefMut will automatically "convert" these.